### ISSUE TITLE
magithub-clone should handle cloning forks

### ISSUE THREAD

**Issue by vermiculus** [vermiculus/magithub#131]
From #130. When cloning a fork, we should in fact clone the source repository and then add the fork as a remote (and set up defaults, etc.) One question that remains is whether to use the `parent` repo (that repo from which the current repo was forked) or the `source` repo (that repo from which all forks originate).

**Comment by stardiviner** [comment]
I think we should treat original parent repo as `upstream` remote.

**Comment by vermiculus** [comment]
@stardiviner I think that's the plan. I *really* need to get back to work on this project :wink: There's handling for this now; I just need to polish and publish.

**Comment by vermiculus** [comment]
Coming back to this since it came up while writing documentation.  Is there a strong reason for not cloning the `source` repository and then adding the fork as a normal remote?  It would work with the other tools a *lot* better.

**Comment by stardiviner** [comment]
My most working forks are Emacs packages, on very little case, if the original author late to accept PR, or not maintaining, like `arduino-mode`. I usually will fork, then use `use-package-quelpa` to use my own fork as package source. Almost every case if I want to PR, I will create a fork at first.

**Comment by vermiculus** [comment]
Yes, and your fork is really just a branch of the source to be merged into its base. Is there any difference between having
```
[upstream]
   url = path/to/source.git
[origin]
   url = path/to/fork.git
[branch "new-feature"]
   remote = origin
```
and
```
[origin]
   url = path/to/source.git
[stardiviner]
   url = path/to/fork.git
[branch "new-feature"]
   remote = stardiviner
```
?

**Comment by stardiviner** [comment]
Ah, you're right, no different. I have not realized that. That's a bad habit now. :)
